chains/proposal.go

```diff
@@ -0,0 +1,87 @@
+package chains
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/chainbridge-core/relayer/message"
+	"github.com/ChainSafe/chainbridge-core/types"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+)
+
+func NewProposal(source, destination uint8, depositNonce uint64, resourceId types.ResourceID, data []byte, metadata message.Metadata) *Proposal {
+	return &Proposal{
+		OriginDomainID: source,
+		DepositNonce:   depositNonce,
+		ResourceID:     resourceId,
+		Destination:    destination,
+		Data:           data,
+		Metadata:       metadata,
+	}
+}
+
+type Proposal struct {
+	OriginDomainID uint8            // Source domainID where message was initiated
+	DepositNonce   uint64           // Nonce for the deposit
+	ResourceID     types.ResourceID // change id -> ID
+	Data           []byte
+	Destination    uint8 // Destination domainID where message is to be sent
+	Metadata       message.Metadata
+}
+
+func NewProposalsHash(proposals []*Proposal, chainID int64, verifContract string, bridgeVersion string) ([]byte, error) {
```

Comment:
It would be more correct to call this just ProposalsHash as it doesn't initialize any struct